### ISSUE TITLE
Precompiling module ends with LoadError

### ISSUE THREAD

**Issue by marcosmolla** [JuliaStats/Distributions.jl#648]
Hi, 
I am new to Julia and only wrote a small program (here: https://github.com/marcosmolla/montyhallproblem) 

However, when I try `using Distributions` the process ends with:

```Julia
julia> using Distributions
INFO: Precompiling module Distributions.
ERROR: LoadError: LoadError: InitError: could not load symbol "unif_rand_ptr":
dlsym(0x7fc7db448a30, unif_rand_ptr): symbol not found
Stacktrace:
 [1] __init__() at /Users/marco/.julia/v0.6/Rmath/src/Rmath.jl:47
 [2] _include_from_serialized(::String) at ./loading.jl:157
 [3] _require_from_serialized(::Int64, ::Symbol, ::String, ::Bool) at ./loading.jl:200
 [4] _require(::Symbol) at ./loading.jl:457
 [5] require(::Symbol) at ./loading.jl:398
 [6] include_from_node1(::String) at ./loading.jl:569
 [7] include(::String) at ./sysimg.jl:14
 [8] include_from_node1(::String) at ./loading.jl:569
 [9] include(::String) at ./sysimg.jl:14
 [10] anonymous at ./<missing>:2
during initialization of module Rmath
while loading /Users/marco/.julia/v0.6/StatsFuns/src/rmath.jl, in expression starting on line 6
while loading /Users/marco/.julia/v0.6/StatsFuns/src/StatsFuns.jl, in expression starting on line 228
ERROR: LoadError: Failed to precompile StatsFuns to /Users/marco/.julia/lib/v0.6/StatsFuns.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:703
 [2] _require(::Symbol) at ./loading.jl:456
 [3] require(::Symbol) at ./loading.jl:398
 [4] include_from_node1(::String) at ./loading.jl:569
 [5] include(::String) at ./sysimg.jl:14
 [6] anonymous at ./<missing>:2
while loading /Users/marco/.julia/v0.6/Distributions/src/Distributions.jl, in expression starting on line 6
ERROR: Failed to precompile Distributions to /Users/marco/.julia/lib/v0.6/Distributions.ji.
Stacktrace:
 [1] compilecache(::String) at ./loading.jl:703
 [2] _require(::Symbol) at ./loading.jl:490
 [3] require(::Symbol) at ./loading.jl:398
```

Also, if I try to `Pkg.build("Distributions")`, I receive:
```Julia
julia> Pkg.build("Distributions")
INFO: Building Rmath
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/downloads
INFO: Directory /Users/marco/.julia/v0.6/Rmath/deps/downloads already created
INFO: Downloading file https://github.com/JuliaLang/Rmath-julia/archive/v0.2.0.tar.gz
INFO: Done downloading file https://github.com/JuliaLang/Rmath-julia/archive/v0.2.0.tar.gz
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/src
INFO: Directory /Users/marco/.julia/v0.6/Rmath/deps/src already created
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps
INFO: Directory /Users/marco/.julia/v0.6/Rmath/deps already created
INFO: Path /Users/marco/.julia/v0.6/Rmath/deps/src/Rmath-julia-0.2.0 already created
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/usr/lib
INFO: Directory /Users/marco/.julia/v0.6/Rmath/deps/usr/lib already created
INFO: Changing Directory to /Users/marco/.julia/v0.6/Rmath/deps/src/Rmath-julia-0.2.0
gcc: error trying to exec 'cc1': execvp: No such file or directory
make[1]: *** [bd0.o] Error 1
make: *** [all] Error 2
===============================================================[ ERROR: Rmath ]===============================================================

LoadError: failed process: Process(`make`, ProcessExited(2)) [2]
while loading /Users/marco/.julia/v0.6/Rmath/deps/build.jl, in expression starting on line 42

==============================================================================================================================================

===============================================================[ BUILD ERRORS ]===============================================================

WARNING: Rmath had build errors.

 - packages with build errors remain installed in /Users/marco/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("Rmath")`
 - build a single package by running its `deps/build.jl` script

==============================================================================================================================================
```

I am using Julia 0.6.0 on a Mac. 
Any ideas?

Many thanks in advance, 
Marco

**Comment by andreasnoack** [comment]
Looks like an `Rmath` issue. It fails if you just do `Pkg.build("Rmath")`, right?

**Comment by ararslan** [comment]
Hm, I can't reproduce on macOS 10.12 with the official Julia 0.6.0 binary. `Pkg.build("Rmath")` works for me, as does `using Distributions`.

**Comment by marcosmolla** [comment]
Thanks for the replies! I deleted julia (from Applications folder) and removed .julia/ (rm -rf .julia) and installed it fresh with the .dmg from the website. However, the problem remains, see below:
```julia
              _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: https://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.6.0 (2017-06-19 13:05 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-apple-darwin13.4.0

julia> using Distributions
ERROR: ArgumentError: Module Distributions not found in current path.
Run `Pkg.add("Distributions")` to install the Distributions package.
Stacktrace:
 [1] _require(::Symbol) at ./loading.jl:428
 [2] require(::Symbol) at ./loading.jl:398

julia> Pkg.add("Distributions")
INFO: Initializing package repository /Users/marco/.julia/v0.6
INFO: Cloning METADATA from https://github.com/JuliaLang/METADATA.jl
INFO: Cloning cache of BinDeps from https://github.com/JuliaLang/BinDeps.jl.git
INFO: Cloning cache of Calculus from https://github.com/johnmyleswhite/Calculus.jl.git
INFO: Cloning cache of Compat from https://github.com/JuliaLang/Compat.jl.git
INFO: Cloning cache of DataStructures from https://github.com/JuliaCollections/DataStructures.jl.git
INFO: Cloning cache of Distributions from https://github.com/JuliaStats/Distributions.jl.git
INFO: Cloning cache of PDMats from https://github.com/JuliaStats/PDMats.jl.git
INFO: Cloning cache of QuadGK from https://github.com/JuliaMath/QuadGK.jl.git
INFO: Cloning cache of Rmath from https://github.com/JuliaStats/Rmath.jl.git
INFO: Cloning cache of SHA from https://github.com/staticfloat/SHA.jl.git
INFO: Cloning cache of SpecialFunctions from https://github.com/JuliaMath/SpecialFunctions.jl.git
INFO: Cloning cache of StatsBase from https://github.com/JuliaStats/StatsBase.jl.git
INFO: Cloning cache of StatsFuns from https://github.com/JuliaStats/StatsFuns.jl.git
INFO: Cloning cache of URIParser from https://github.com/JuliaWeb/URIParser.jl.git
INFO: Installing BinDeps v0.7.0
INFO: Installing Calculus v0.2.2
INFO: Installing Compat v0.29.0
INFO: Installing DataStructures v0.6.1
INFO: Installing Distributions v0.14.2
INFO: Installing PDMats v0.7.0
INFO: Installing QuadGK v0.1.3
INFO: Installing Rmath v0.2.0
INFO: Installing SHA v0.5.0
INFO: Installing SpecialFunctions v0.3.0
INFO: Installing StatsBase v0.18.0
INFO: Installing StatsFuns v0.5.0
INFO: Installing URIParser v0.2.0
INFO: Building Rmath
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/downloads
INFO: Downloading file https://github.com/JuliaLang/Rmath-julia/archive/v0.2.0.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   129    0   129    0     0    983      0 --:--:-- --:--:-- --:--:--   984
100  155k  100  155k    0     0   502k      0 --:--:-- --:--:-- --:--:--  502k
INFO: Done downloading file https://github.com/JuliaLang/Rmath-julia/archive/v0.2.0.tar.gz
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/src
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps
INFO: Directory /Users/marco/.julia/v0.6/Rmath/deps already created
INFO: Attempting to Create directory /Users/marco/.julia/v0.6/Rmath/deps/usr/lib
INFO: Changing Directory to /Users/marco/.julia/v0.6/Rmath/deps/src/Rmath-julia-0.2.0
gcc: error trying to exec 'cc1': execvp: No such file or directory
make[1]: *** [bd0.o] Error 1
make: *** [all] Error 2
====================================================[ ERROR: Rmath ]====================================================

LoadError: failed process: Process(`make`, ProcessExited(2)) [2]
while loading /Users/marco/.julia/v0.6/Rmath/deps/build.jl, in expression starting on line 42

========================================================================================================================

====================================================[ BUILD ERRORS ]====================================================

WARNING: Rmath had build errors.

 - packages with build errors remain installed in /Users/marco/.julia/v0.6
 - build the package(s) and all dependencies with `Pkg.build("Rmath")`
 - build a single package by running its `deps/build.jl` script

========================================================================================================================
INFO: Package database updated

julia> 
```

**Comment by marcosmolla** [comment]
Okay, it works now. I restarted julia, this time I first ran `Pkg.build("Rmath")` and then `using Distributions`This time it ran smoothly. Not sure what exactly the reason is. Though, I did change my `$PATH` from `usr/bin/local:usr/bin:` to `usr/bin:usr/bin/local:`. A colleague told me it could be that it is looking for the `gcc` in the wrong place. Not sure, whether this was actually what solved it. 
Anyway, thanks for your replies again!
